KarhooSDK/Service/User/KarhooPaymentProviderUpdateHandler.swift

```diff
@@ -0,0 +1,59 @@
+//
+// Created by Bartlomiej Sopala on 18/07/2022.
+//
+
+import Foundation
+
+class KarhooPaymentProviderUpdateHandler: PaymentProviderUpdateHandler {
+
+    private let userDataStore: UserDataStore
+    private let nonceRequestSender: RequestSender
+    private let paymentProviderRequest: RequestSender
+    private let loyaltyProviderRequest: RequestSender
+
+    init(
+        userDataStore: UserDataStore = DefaultUserDataStore(),
+        nonceRequestSender: RequestSender = KarhooRequestSender(httpClient: TokenRefreshingHttpClient.shared),
+        paymentProviderRequest: RequestSender = KarhooRequestSender(httpClient: TokenRefreshingHttpClient.shared),
```

Comment:
this is sometimes JsonHttpClient, other times TokenRefreshHttpClient. let's keep the same logic as before and pass the instantiated value every time.